pkg/operator/staticpod/controller/startupmonitorcondition/startup_monitor_condition.go

```diff
@@ -46,21 +49,24 @@ func New(targetNamespace string,
 			operatorClient.Informer(),
 		).
 		ToController(
-			"StartupMonitorPodCondition", // don't change what is passed here unless you also remove the old FooDegraded condition
+			fd.controllerInstanceName,
 			eventRecorder,
 		)
 }
 
 func (fd *startupMonitorPodConditionController) sync(ctx context.Context, _ factory.SyncContext) (err error) {
 	startupPodDegraded := &operatorv1.OperatorCondition{Type: "StartupMonitorPodDegraded", Status: operatorv1.ConditionFalse}
 	startupPodContainerExcessiveRestartsDegraded := &operatorv1.OperatorCondition{Type: "StartupMonitorPodContainerExcessiveRestartsDegraded", Status: operatorv1.ConditionFalse}
+	status := applyoperatorv1.OperatorStatus()
 
 	defer func() {
 		if err == nil {
-			if _, _, updateError := operatorv1helpers.UpdateStatus(ctx, fd.operatorClient,
-				operatorv1helpers.UpdateConditionFn(*startupPodDegraded),
-				operatorv1helpers.UpdateConditionFn(*startupPodContainerExcessiveRestartsDegraded)); updateError != nil {
-				err = updateError
+			status = status.WithConditions(
+				conditionToApplyOperatorCondition(startupPodDegraded),
```

Comment:
No.  Switch the conditions themselves to applyconfiguration types so we avoid accidents due to conversion.

Reply:
Good point, reworked the PR.